src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,133 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Bias().begin(),
+      module->Bias().end(), module->Parameters().end() - outSize,
+      module->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(module->Weight().n_rows, kernelWidth);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_cols, kernelHeight);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_slices, inSize * outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_cols, 1);
+  BOOST_REQUIRE_EQUAL(module->Parameters().n_rows,
+      (outSize * inSize * kernelWidth * kernelHeight) + outSize);
+
+  delete module;
+}
+
+/**
+ * Convolution module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(ConvolutionLayerWeightInitializationTest)
+{
+  size_t inSize = 2, outSize = 2;
+  size_t kernelWidth = 3, kernelHeight = 3;
+  Convolution<>* module = new Convolution<>(inSize, outSize,
+      kernelWidth, kernelHeight);
+  module->Reset();
+  RandomInitialization().Initialize(module->Weight(), kernelWidth, kernelHeight,
+      inSize * outSize);
+  module->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Weight().begin(),
+      module->Weight().end(), module->Parameters().begin(),
+      module->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Bias().begin(),
+      module->Bias().end(), module->Parameters().end() - outSize,
+      module->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(module->Weight().n_rows, kernelWidth);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_cols, kernelHeight);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_slices, inSize * outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_cols, 1);
+  BOOST_REQUIRE_EQUAL(module->Parameters().n_rows,
+      (outSize * inSize * kernelWidth * kernelHeight) + outSize);
+
+  delete module;
+}
+
+/**
+ * Transposed Convolution module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(TransposedConvolutionLayerWeightInitializationTest)
+{
+  size_t inSize = 1, outSize = 1;
+  size_t kernelWidth = 4, kernelHeight = 4;
+  TransposedConvolution<>* module = new TransposedConvolution<>(inSize, outSize,
+      kernelWidth, kernelHeight, 1, 1, 1, 1, 5, 5, 6, 6);
+  module->Reset();
+  RandomInitialization().Initialize(module->Weight(), kernelWidth, kernelHeight,
+      inSize * outSize);
+  module->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Weight().begin(),
+      module->Weight().end(), module->Parameters().begin(),
+      module->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Bias().begin(),
+      module->Bias().end(), module->Parameters().end() - outSize,
+      module->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(module->Weight().n_rows, kernelWidth);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_cols, kernelHeight);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_slices, inSize * outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_rows, 1);
```

Comment:
Looks like this is an error. It could be easy to find if you used different values for `inSize`, `outSize`, `kernelWidth`, and `kernelHeight`.
```suggestion
  BOOST_REQUIRE_EQUAL(module->Bias().n_cols, 1);
```